src/shared_utilities/utils.jl

```diff
@@ -542,6 +564,36 @@ function count_nans_state(
     return nothing
 end
 
+"""
+    count_nans_state(
+        state::ClimaCore.Fields.Field,
+        space::ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace;
+        mask = nothing,
+        verbose = false,
+    )
+
+Counts the NaNs in the surface level of the field `state`,
+ where the `state is defined on an extruded finite difference space.
+"""
+function count_nans_state(
+    state::ClimaCore.Fields.Field,
+    space::ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace;
+    mask = nothing,
+    verbose = false,
+)
+    # Note: this code uses `parent`; this pattern should not be replicated
+    surface_state = ClimaLand.Domains.top_center_to_surface(state)
+    num_nans =
+        isnothing(mask) ? round(sum(isnan, parent(surface_state))) :
+        round(sum(isnan, parent(surface_state)[Bool.(parent(mask))]; init = 0))
```

Comment:
@juliasloan25 does this need to be a regular array to support the indexing?  not sure why that would affect the gpu long runs, though, and not the GPU unit tests.